arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -132,6 +132,7 @@ void RocksDBReplicationContext::removeVocbase(TRI_vocbase_t& vocbase) {
     if (it->second->vocbase.id() == vocbase.id()) {
       if (it->second->isUsed()) {
         LOG_TOPIC("543d4", ERR, Logger::REPLICATION) << "trying to delete used context";
+        it++;
```

Comment:
are we generating leftover here?

The original code did run into an "endless" loop waiting until all contexts are unused (there are parallel threads working on them) the new code simply jumps over the used contexts, and returns from this function (there is no report of this operation was a success or not)

Reply:
I think the original code was not correct. It would not make any progress and block the thread for a potential endless amount of time.